libs/platform/ebpf_epoch.c

```diff
@@ -651,70 +663,20 @@ _ebpf_epoch_get_release_epoch(_Out_ int64_t* release_epoch)
             lowest_epoch = min(lowest_epoch, _ebpf_epoch_cpu_table[cpu_id].epoch_state.epoch);
         }
 
-        // If a thread was last used before this cutoff, then it is stale and should be removed.
-        uint64_t stale_thread_cutoff =
-            now - (EBPF_EPOCH_THREAD_TABLE_TIMEOUT_IN_NANO_SECONDS / EBPF_NANO_SECONDS_PER_FILETIME_TICK);
-
-        // Loop over all the threads in this CPU entry.
-        do {
-            // Get the next per-thread entry from this CPU.
-            return_value = ebpf_hash_table_next_key_and_value(
-                _ebpf_epoch_cpu_table[cpu_id].thread_table,
-                thread_id == 0 ? NULL : (uint8_t*)&thread_id,
-                (uint8_t*)&thread_id,
-                (uint8_t**)&thread_entry);
-
-            // There are no more entries in the table.
-            if (return_value != EBPF_SUCCESS) {
-                break;
-            }
-
-            // Include this epoch state if it's active.
-            if (thread_entry->epoch_state.active) {
-                int64_t age = now - thread_entry->last_used_time;
-                if (age > EBPF_EPOCH_STALE_THREAD_TIME_IN_NANO_SECONDS) {
-                    EBPF_LOG_MESSAGE_UINT64_UINT64(
-                        EBPF_TRACELOG_LEVEL_VERBOSE,
-                        EBPF_TRACELOG_KEYWORD_EPOCH,
-                        "Stale active thread entry",
-                        (uint64_t)thread_id,
-                        age);
-                    // Reset last_used_time time to limit rate of logging.
-                    thread_entry->last_used_time = now;
-                }
+        // Loop over the thread table and compute the lowest active epoch.
+        for (size_t index = 0; index < EBPF_EPOCH_THREAD_ENTRY_TABLE_SIZE; index++) {
+            thread_entry = &_ebpf_epoch_cpu_table[cpu_id].thread_entry_table[index];
```

Comment:
A macro for these 2 lines would be useful since its in 3 places.

Reply:
Two instances were replaced with helper function, but this one doesn't match the same pattern.